core/src/main/scala/kafka/log/LogValidator.scala

```diff
@@ -234,17 +234,16 @@ private[log] object LogValidator extends Logging {
 
     val firstBatch = getFirstBatchAndMaybeValidateNoMoreBatches(records, NoCompressionCodec)
 
-    records.batches.forEach { batch =>
+    for (batch <- records.batches.asScala) {
```

Comment:
I liked your changes to make the code more concise, I'd keep them.

Reply:
ok~